scripts/pylib/twister/twisterlib/statuses.py

```diff
@@ -0,0 +1,91 @@
+#!/usr/bin/env python3
+# Copyright (c) 2024 Intel Corporation
+#
+# SPDX-License-Identifier: Apache-2.0
+"""
+Status classes to be used instead of str statuses.
+"""
+
+from enum import Enum
+
+
+# Status Assignment tree:
+#   parent <- child <=> parent is assigned to child in code.
+#
+#   HarnessStatus
+#   └ QEMUOutputStatus
+#     └ TestInstanceStatus
+#       ├ TestSuiteStatus
+#       └ TestCaseStatus
+
+
+class TestInstanceStatus(str, Enum):
+    def __str__(self):
+        return str(self.value)
+
+    # Values assigned directly
+    NONE = None  # to preserve old functionality
+    ERROR = 'error'
+    FAIL = 'failed'
+    FILTER = 'filtered'
+    PASS = 'passed'
+    SKIP = 'skipped'
+
+
+class TestSuiteStatus(str, Enum):
```

Comment:
why not just do below, the reasons are:
1. rephase the enumrate from before seems confusing.
2. give a way that we can just judge the return value to see it pass and failed
`
class TestSuiteStatus(str, Enum):
    none = None
    passed = 0
    filtered = 1
    skipped = 2
    error = 3
    failed = 4

    def __str__(self):
        return str(self.name)

`

Reply:
There are possible problems with both `none` and `passed` being falsey while the others are truthy.

Technically it would be a change to results, as the old "None" statuses would now be "none" statuses (Minor, could be changed by a check in `__str__`, but still warrants a mention).

This would tie the status `str` representation strongly to the code representation, thus making further changes harder (What if someone decides that the fact that all statuses use past participle save for `error` is unacceptable and it should be changed to `errored`? We would need to change every use of the `TestSuiteStatus.error`.)

We currently have no use for checking just the return value instead of the status name. Doing so would make the code less readable (Compare: `status in [Status.SKIP, Status.FILTER]` vs `return_code in [1, 2]`)